storops/connection/client.py

```diff
@@ -110,7 +110,11 @@ def request(self, full_url, method, **kwargs):
             except ValueError:
                 pass
 
-        if resp.status_code == 401:
+        # The reason why not raise if `status_code` >= 400 is that we don't
+        # want to raise if code is 404 - Not Found. The upper layer will check
+        # the returned resource and raise `ResourceNotFoundError` or some thing
+        # like it.
+        if resp.status_code >= 400 and resp.status_code != 404:
```

Comment:
So this error may also happened with other error codes, like 409 or 422. 
I thought you may check good status code like 200, 201, 202, and 204, and all the others would be considered as error response.
The change looks good as well.